python/plugins/processing/algs/qgis/RasterCalculator.py

```diff
@@ -157,15 +160,23 @@ def _cellsize(layer):
         if cellsize == 0:
             cellsize = min([_cellsize(lyr) for lyr in layersDict.values()])
 
+        # check for layers available in the model
+        layersDictCopy = layersDict.copy() # need a shallow copy because next calls invalidate iterator
+        for lyr in layersDictCopy.values():
+            expression = self.mappedNameToLayer(lyr, expression, layersDict, context)
```

Comment:
This is not adding all layers that are available in the model, it's adding layers that are added as '*Reference layer(s) (used for automated extent, cellsize, and CRS) [optional]*' (`self.LAYERS`). Any intermediary layers that are not added will be ignored in the calculator (which will be very confusing for users).

Reply:
I sincerly don't want the "upper" logic of the processing algorithm... my scope here is only to manage layer in self.LAYERS if any or that are passed in expression scope. I self.LAYERS is not filled as expected is out the scope of this PR or fix.